src/api/app/models/kiwi/image/xml_builder.rb

```diff
@@ -14,12 +14,56 @@ def build
         doc = update_packages(doc)
         doc = update_repositories(doc)
         doc = update_description(doc)
+        doc = update_preferences(doc)
 
         Nokogiri::XML(doc.to_xml, &:noblanks).to_xml(indent: kiwi_indentation(doc))
       end
 
       private
 
+      def update_preferences(document)
```

Comment:
I don't like the comments in this method.

Btw: I think this method is so complex because we "randomly" pick attributes of the image and implement them instead of doing it a little bit more thought-out. Nothing wrong with your PR @evanrolfe as you just implement what was in the sprint backlog. I just say that we failed in the grooming to not raise concerns about this feature 😢 
If we would instead implement the different configurations step by step I assume this wouldn't be that complicated as it is right now!

Reply:
Maybe it makes more sense to describe the algorithm and an example XML as header of the method instead of the inline comments. For instance:

```
  Updates the containerconfig name and tag. Creates it if it doesn't exist!
    <preferences>
      <type image="docker" boot="grub">
        <containerconfig name="my_container" tag="latest"/>
        <oemconfig>test</oemconfig>
      </type>
      <bootsplash-theme>gnome</bootsplash-theme>
      <bootloader-theme>gnome-dark</bootloader-theme>
    </preferences>
```

Reply:
> ChrisBr 17 hours ago 
>
> Nothing wrong with your PR @evanrolfe as you just implement what was in the sprint backlog. I just say that we failed in the grooming to not raise concerns about this feature 😢

@ChrisBr We can still raise concerns and talk to Adrian, if we have any doubts